### DISCUSSION TITLE
report "App preview failed to load An unknown error occurred" how to fix it or debug it?

### DISCUSSION THREAD

**Discussion by cwmine** [Budibase/budibase#4058] [Help] [answered]
I link a mysql table and display normally in data/source
In Design/Screen, I add a AUTOGENERATED SCREENS, the preview page display  "App preview failed to load An unknown error occurred" .
Any ideal to fix it or find what's wrong in the operations?

**Comment by mjashanks** [comment]
Hi @cwmine - I could not reproduce this issue - I successfully autogenerated screens from a MySQL Table.

Can you..
1. Provide a screen shot of the error displayed on the preview screen, as you mentioned
2. Open up your browser's dev tools, and re-add the autogenerated screen. Are there any errors in your console? Could you provide a screenshot of this?

Thanks!

**Comment by cwmine** [comment]
Hi @mjashanks 
Edge Console prompts:
Failed to load resource: the server responded with a status of 400 (Bad Request)
budibase-client.js:1 Failed to load resource: the server responded with a status of 400 (Bad Request)

this is screen clip:

Devtools:


when I access the error js file (budibase-client.js) in the browser, the error is displayed as below:
![uTools_1642576238384](https://user-images.githubusercontent.com/4034274/150081645-76b7d539-1ead-4775-a583-d1535d714cf1.png)

**Comment by shogunpurple** [comment]
Hi @cwmine - I think I know what is wrong here. I thought originally that `signatureVersion: v4` wasn't being set, but it is hardcoded in our S3 configuration for the node SDK.

Is your MinIO instance running? It seems like this error is coming directly from AWS and it's trying to access a bucket that you don't own. 

How did you set up budibase? Have you changed any of the default configuration around MinIO?

**Comment by cwmine** [comment, marked as answer]
@shogunpurple Hi Thanks for your replying.
My MinIo instance is running without any erorr prompted. and I do not modify any seetings. 
However, I found this bug only occurs when I access through my Apache proxy. (in https). When I access the budibase site directly using the localhost port, it runs normally.
This is my Apache proxy setting as below. Are there some mistakes in it?

```
 <VirtualHost *:443>
ServerName budi.xxx.xx
   SSLProxyEngine On
   SSLProxyCheckPeerCN Off
   SSLProxyCheckPeerName Off
   SSLProxyVerify none
ProxyPass / http://127.0.0.1:11090/
ProxyPassReverse / http://127.0.0.1:11090/
ProxyPreserveHost On
</VirtualHost>
```

**Reply by shogunpurple** [reply]
Hey @cwmine - that apache configuration looks good to me. What I'm wondering is that maybe it's not passing headers correctly from the proxy to budibase. Do you have the `mod_proxy_http` module enabled?

Also not sure if it matters in apache, but it seems like your indentation is not correct under your `ServerName`. It should be:

```
<VirtualHost *:443>
ServerName budi.svn.os
   SSLProxyEngine On
   SSLProxyCheckPeerCN Off
   SSLProxyCheckPeerName Off
   SSLProxyVerify none
   ProxyPass / http://127.0.0.1:11090/
   ProxyPassReverse / http://127.0.0.1:11090/
   ProxyPreserveHost On
</VirtualHost>
```

**Reply by cwmine** [reply]
@shogunpurple Hi
I have finally found the source of the bug. My apache base site is autherized by HTTP basic autherization. and when I access the site and the proxy to budibase. The header of request will be sent with header data "autherizaion: base XXXX" . It seems to confict with the Minio autherization mechanism. 

And I fix it by adding `Require all granted` to the Apache proxy configurations.

Thanks for your help.

**Reply by shogunpurple** [reply]
Awesome! Glad to hear it. Let us know if you run into anything else!

**Comment by duyan11110** [comment]
Hi,
I have the same error when designing. Here's error detail:
![image](https://user-images.githubusercontent.com/10897328/162114853-34c09078-69d1-4804-a22e-a9534c913902.png)
![image](https://user-images.githubusercontent.com/10897328/162115000-ebebca4a-b4db-4bfa-b0c3-4b0f60d0f13a.png)

I deployed Budibase version 1.0.108 on Docker for Window. Docker version 20.10.13, build a224086.
Pls help me how to fix it.
